doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -144,61 +144,142 @@ presentation of Cic is given in Chapter :ref:`calculusofinductiveconstructions`.
-   qualid           : `ident` | `qualid` `field`
-   sort             : SProp | Prop | Set | Type
-   fix_bodies       : `fix_body`
-                    : `fix_body` with `fix_body` with … with `fix_body` for `ident`
-   cofix_bodies     : `cofix_body`
-                    : `cofix_body` with `cofix_body` with … with `cofix_body` for `ident`
-   fix_body         : `ident` `binders` [`annotation`] [: `term`] := `term`
-   cofix_body       : `ident` [`binders`] [: `term`] := `term`
-   annotation       : { struct `ident` }
-   match_item       : `term` [as `name`] [in `qualid` [`pattern` … `pattern`]]
-   dep_ret_type     : [as `name`] `return_type`
-   return_type      : return `term`
-   equation         : `mult_pattern` | … | `mult_pattern` => `term`
-   mult_pattern     : `pattern` , … , `pattern`
-   pattern          : `qualid` `pattern` … `pattern`
-                    : @ `qualid` `pattern` … `pattern`
-                    : `pattern` as `ident`
-                    : `pattern` % `ident`
-                    : `qualid`
-                    : _
-                    : `num`
-                    : ( `pattern` | … | `pattern` )
+   term : forall `open_binders` , `term`
+        : fun `open_binders` => `term`
+        : let `name` `binder_list_opt` `type_cstr` := `term` in `term`
+        : let `single_fix` in `term`
+        : let `name_alt` `return_type_opt` := `term` in `term`
+        : let ' `pattern` := `term` in `term`
+        : let ' `pattern` := `term` return `term100` in `term`
+        : let ' `pattern` in `pattern` := `term` return `term100` in `term`
+        : if `term` `return_type_opt` then `term` else `term`
+        : `single_fix`
+        : `single_fix` with `fix_body_list` for `ident`
+        : `term100`
+   type_cstr : : `term`
+             : `empty`
+   name : _
+        : `ident`
+   name_alt : ( `name_list_comma_opt` )
+            : ()
+   name_list_comma_opt : `name_list_comma`
+                       : `empty`
+   name_list_comma : `name_list_comma` , `name`
+                   : `name`
+   open_binders : `name` `name_list_opt` : `term`
+                : `name` `name_list_opt` `binder_list_opt`
+                : `name` .. `name`
+                : `closed_binder` `binder_list_opt`
+   name_list_opt : `name_list_opt` `name`
+                 : `empty`
+   binder_list_opt : `binder_list_opt` `binder`
+                   : `empty`
+   binder : `name`
+          : `closed_binder`
+   closed_binder : ( `name` `name_list` : `term` )
+                 : ( `name` : `term` )
+                 : ( `name` := `term` )
+                 : ( `name` : `term` := `term` )
+                 : { `name` }
+                 : { `name` `name_list` : `term` }
+                 : { `name` : `term` }
+                 : { `name` `name_list` }
+                 : `( `typeclass_constraint_list_comma` )
+                 : `{ `typeclass_constraint_list_comma` }
+                 : ' `pattern0`
+                 : ( _ : `term` | `term` )
```

Comment:
I would prefer to keep the part on binders in a separate section like you did initially, despite having inlined `binder_constr`.

Reply:
ok